tools/remap/convert_remap_actions_to_10x

```diff
@@ -0,0 +1,172 @@
+        >>> convert_line('map http://example.com http://backend.example.com @action=add_allow @method=GET')
+        'map http://example.com http://backend.example.com @action=add_allow @method=GET'
+
+        # Comments should be converted too.
+        >>> convert_line('# Using @action=allow is nice.')
+        '# Using @action=add_allow is nice.'
+        >>> convert_line('# map http://example.com http://backend.example.com @action=allow @method=GET')
+        '# map http://example.com http://backend.example.com @action=add_allow @method=GET'
+    '''
+    output_line = input_line.replace('@action=allow', '@action=add_allow')
+    output_line = output_line.replace('@action=deny', '@action=add_deny')
+    return output_line
+
+
+def copy_input_file(input: str, no_backup: bool) -> str:
+    '''Create a copy of the input file.
+
+    :param input: The input file path from which to create a copy.
+
+    :param no_backup: Do not create a backup file from input. Rather simply
+    create a temporary file that will be later removed. Otherwise, create a
+    backup file of the format <input>.action_to_add.bak.<num>, where <num> is
+    one greater than the greatest value in the directory where @a input exists.
+
+    :return: The path to the copied file.
+    '''
+    if no_backup:
+        copy_path = tempfile.NamedTemporaryFile(delete=False).name
+    else:
+        input_dir = os.path.dirname(input)
+        input_dir = '.' if input_dir == '' else input_dir
+        input_base = os.path.basename(input)
+        # Find the largest <num> in input_dir/input_base.action_to_add.bak.<num>.
+        max_num = 0
+        for file in os.listdir(input_dir):
+            if file.startswith(f'{input_base}.action_to_add.bak.'):
+                try:
+                    num = int(file.split('.')[-1])
+                except ValueError:
+                    # The suffix is not a number. Ignore this file.
+                    continue
+                max_num = max(max_num, num)
+        copy_path = os.path.join(input_dir, f'{input_base}.action_to_add.bak.{max_num + 1}')
+    shutil.copyfile(input, copy_path)
+    return copy_path
+
+
+def prepare_files(input: str, output: Optional[str], no_backup: bool) -> Tuple[str, str]:
+    '''Prepare the input and output files.
+
+    :param input: The input file path.
+    :param output: The output file path.
+    :param no_backup: Do not create a backup of the input file when <input> is modified.
+
+    :return: A tuple containing the input and output filenames, in that order.
+    The input file will contain the content from which to copy, the output file
+    will reference a file whose content should be overridden. The caller should
+    delete the input filename if @a no_backup is true. Otherwise the input
+    filename will be the path to the backup file.
+    '''
+    if not os.path.exists(input):
+        raise FileNotFoundError(f'Input file does not exist: {input}')
+
+    if input == output:
```

Comment:
Uses a temporary file to make the changes before replacing the output. Very good. Comment might not hurt but not necessary either.

Reply:
The `copy_input_file` doc says it will create a temporary. Would you suggest another comment here?

Reply:
Nah.